lib/ansible/modules/remote_management/hpe/oneview_fc_network.py

```diff
@@ -0,0 +1,138 @@
+version_added: "2.4"
+requirements:
+    - "hpOneView >= 4.0.0"
+author: "Felipe Bulsoni (@fgbulsoni)"
+options:
+    state:
+        description:
+            - Indicates the desired state for the Fibre Channel Network resource.
+              C(present) will ensure data properties are compliant with OneView.
+              C(absent) will remove the resource from OneView, if it exists.
+        choices: ['present', 'absent']
+    data:
+        description:
+            - List with the Fibre Channel Network properties.
+        required: true
+
+extends_documentation_fragment:
+    - oneview
+    - oneview.validateetag
+'''
+
+EXAMPLES = '''
+- name: Ensure that the Fibre Channel Network is present using the default configuration
+  oneview_fc_network:
+    config: "{{ config_file_path }}"
+    state: present
+    data:
+      name: 'New FC Network'
+
+- name: Ensure that the Fibre Channel Network is present with fabricType 'DirectAttach'
+  oneview_fc_network:
+    config: "{{ config_file_path }}"
+    state: present
+    data:
+      name: 'New FC Network'
+      fabricType: 'DirectAttach'
+
+- name: Ensure that the Fibre Channel Network is present and is inserted in the desired scopes
+  oneview_fc_network:
+    config: "{{ config_file_path }}"
+    state: present
+    data:
+      name: 'New FC Network'
+      scopeUris:
+        - '/rest/scopes/00SC123456'
+        - '/rest/scopes/01SC123456'
+
+- name: Ensure that the Fibre Channel Network is absent
+  oneview_fc_network:
+    config: "{{ config_file_path }}"
+    state: absent
+    data:
+      name: 'New FC Network'
+'''
+
+RETURN = '''
+fc_network:
+    description: Has the facts about the managed OneView FC Network.
+    returned: On state 'present'. Can be null.
+    type: dict
+'''
+
+from ansible.module_utils.oneview import OneViewModuleBase
+
+
+class FcNetworkModule(OneViewModuleBase):
+    MSG_CREATED = 'FC Network created successfully.'
+    MSG_UPDATED = 'FC Network updated successfully.'
+    MSG_DELETED = 'FC Network deleted successfully.'
+    MSG_ALREADY_PRESENT = 'FC Network is already present.'
+    MSG_ALREADY_ABSENT = 'FC Network is already absent.'
+    RESOURCE_FACT_NAME = 'fc_network'
+
+    def __init__(self):
+
+        additional_arg_spec = dict(data=dict(required=True, type='dict'),
+                                   state=dict(
+                                       required=True,
+                                       choices=['present', 'absent']))
+
+        super(FcNetworkModule, self).__init__(additional_arg_spec=additional_arg_spec,
+                                              validate_etag_support=True)
+
+        self.resource_client = self.oneview_client.fc_networks
+
+    def execute_module(self):
+        resource = self.get_by_name(self.data['name'])
+
+        if self.state == 'present':
+            return self.__present(resource)
+        else:
+            return self.resource_absent(resource)
+
+    def __present(self, resource):
```

Comment:
There's still a few of these that are using double-underscore to mark private.  We generally do that with single underscore instead.

``` python
- def __present(self,  resource):
+ def _present(self,  resource):
```